migration/codec.proto

```diff
@@ -0,0 +1,23 @@
+syntax = "proto3";
+
+package migration;
+
+import "github.com/iov-one/weave/codec.proto";
+
+// Schema declares the maxiumum supported schema version for a package.
+message Schema {
+  weave.Header header = 1;
+  // Pkg holds the name of the package that this migration is declared for.
+  // For example, for extension `x/myext` package value is `myext`
+  string pkg = 2;
+  // Version holds the highest supported schema version.
+  uint32 version = 3;
+}
+
+// UpgradeSchemaMsg is a request to upgrade schema version of a given package
```

Comment:
Would there ever be a reason to allow multiple versions at once?

Genesis file should support eg. start at 3 if that was the most recent code when the chain was launched, but I guess it makes sense to require each upgrade to be step by step

Reply:
I hope not. That is making the system harder to deal with.

I think that in the case of genesis, you will always point to schema version 1 and only after loading the genesis you can start migrating data and schemas.